### PR TITLE
manicode: init at 1.0.94

### PR DESCRIPTION
**Pull request by malob**
Add new package `manicode`.

Had to manually include the `package-lock.json`, since it's not included in the source, and there's no upstream repo to request changes to. However, I added an `update.sh` script which generates the `package-lock.json` file, to make updates easier.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ma/manicode/package.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  buildNpmPackage,
+  fetchzip,
+}:
+
+buildNpmPackage rec {
+  pname = "manicode";
+  version = "1.0.94";
+
+  src = fetchzip {
+    url = "https://registry.npmjs.org/${pname}/-/${pname}-${version}.tgz";
```
Comment (Aleksanaa) [inline pkgs/by-name/ma/manicode/package.nix:12]:
```suggestion
    url = "https://registry.npmjs.org/manicode/-/manicode-${version}.tgz";
```

File: pkgs/by-name/ma/manicode/package.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  dontNpmBuild = true;
+
+  passthru.updateScript = ./update.sh;
+
+  meta = with lib; {
+    description = "Use natural language to edit your codebase and run commands from your terminal faster";
+    homepage = "https://manicode.ai";
+    downloadPage = "https://www.npmjs.com/package/manicode";
+    license = licenses.mit;
+    maintainers = [ maintainers.malo ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ma/manicode/package.nix:31]:
```suggestion
  meta = {
    description = "Use natural language to edit your codebase and run commands from your terminal faster";
    homepage = "https://manicode.ai";
    downloadPage = "https://www.npmjs.com/package/manicode";
    license = lib.licenses.mit;
    maintainers = [ lib.maintainers.malo ];
```

Also missing `mainProgram`

File: pkgs/by-name/ma/manicode/package.nix (hunk 1 above)
Comment (malob) [reply pkgs/by-name/ma/manicode/package.nix]: Done

File: pkgs/by-name/ma/manicode/package.nix (hunk 2 above)
Comment (malob) [reply pkgs/by-name/ma/manicode/package.nix]: Huh, I thought `with lib;` was the preferred pattern. Never really been a big fan of that. Happy to make the change. Done.

**Pull request merged**